Jenkinsfile

```diff
@@ -0,0 +1,32 @@
+def docker_images = ["librecores/hosting", " librecores/librecores-ci-modules-fpga", "librecores/ci-modules" , "librecores/librecores-ci-riscv", "librecores/librecores-ci"]
+
+def get_stages(docker_image) {
+    stages = {
+        docker.image(docker_image).inside {
+            stage("${docker_image}") {
+                echo 'Running in ${docker_image}'
+            }
+            stage("Build") {
+                sh 'echo this is stage Build'
+                container = docker.build('docker_image')
```

Comment:
There should be proper tags to avoid race conditions

Reply:
@oleg-nenashev, As discussed I have commented Stage Test for doing it later on. I have added some changes to avoid race conditions and to enable retention strategy